Source/Core/DolphinQt2/MenuBar.cpp

```diff
@@ -206,3 +215,27 @@ void MenuBar::AddTableColumnsMenu(QMenu* view_menu)
     action->setCheckable(true);
   }
 }
+
+void MenuBar::InstallWAD()
+{
+  QString wad_file = QFileDialog::getOpenFileName(this, tr("Select a WAD to install to NAND"),
+                                                  QString(), tr("WAD files (*.wad)"));
+
+  if (wad_file.isEmpty())
+    return;
+
+  QMessageBox result_dialog(this);
+
+  if (GameFile(wad_file).Install())
+  {
+    result_dialog.setIcon(QMessageBox::Information);
+    result_dialog.setText(tr("Successfully installed the WAD to NAND"));
+  }
+  else
+  {
+    result_dialog.setIcon(QMessageBox::Critical);
+    result_dialog.setText(tr("Failed to install the WAD to NAND!"));
+  }
+
+  result_dialog.exec();
+}
```

Comment:
Newline can't be added because the linter doesn't like it. Very weird, but there seems to be no way around it for the moment.